### PR TITLE
Draft for git appendix

### PR DESCRIPTION
**Pull request by dvberkel**
As discussed in issue #64 I created a rough draft for a appendix about using git.

It should fit in right after `Automation with Make` and shows you how to use git to add and commit files.

### CODE REVIEW

**Comment by dvberkel** [comment]
@Charlotteis I am especially interested in your thoughts, tips and tricks for the draft. It would be great if you could review it.

File: src/appendix/git.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+With the content
+
+```text
+build
+```
+
+The `git status` command now does not list `build/` anymore. Instead is lists
+the newly created `.gitignore` file.
+
+## Rince and Repeat
```
Comment (dariusk) [inline src/appendix/git.md]: Rince -> Rinse

Comment (dvberkel) [inline src/appendix/git.md]: @dariusk thanks, will rectify that

File: src/appendix/git.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
+# Appendix C: Version control
+
+Some early computer games had no means of saving your progress. So when you
+painstakingly dodged spear, evaded boulders, jumped crevasses and made a
```
Comment (varjmes) [inline src/appendix/git.md]: spear -> spears

File: src/appendix/git.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+mistake, you had to start all over again. Saving your progress makes playing
+games easier. Version control is like saving your progress for projects.
```
Comment (varjmes) [inline src/appendix/git.md:6]: I _love_ this analogy.

File: src/appendix/git.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+In this appendix we will cover git and how you can use it keep track of changes
+in your project.
+
+## Installing `git`
+
+We use the version control system [`git`][git]. In this section you can find
+details how to setup git for your system.
```
Comment (varjmes) [inline src/appendix/git.md]: 'details how to' -> 'details on how to'
'setup' -> 'set up'

File: src/appendix/git.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+### Mac OS X
+
+There is a binary installer for `git` that can be found at
+[http://git-scm.com/download/mac](http://git-scm.com/download/mac).
+
+### Windows
+
+I hope to provide better instructions for windows soon. If you know how consider
+[contributing][] to the book.
```
Comment (varjmes) [inline src/appendix/git.md:33]: need to put a link here for the contributing

File: src/appendix/git.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+An editor will open so you can provide a _commit message_, a description of the
+commit that will help in identifying what changed. For example,
+
+```text
+A minimal OS that outputs "Hello, World!"
+```
+
+## Ignoring
```
Comment (varjmes) [inline src/appendix/git.md]: I wonder if this should be the first thing people do before learning to add files. Impatient people like me would likely have already added and committed files I wanted to ignore :scream:

File: src/appendix/git.md (hunk 2 above)
Comment (dvberkel) [inline src/appendix/git.md]: @Charlotteis thanks, will change that

File: src/appendix/git.md (hunk 3 above)
Comment (dvberkel) [inline src/appendix/git.md:6]: I am quite fond of it myself

File: src/appendix/git.md (hunk 4 above)
Comment (dvberkel) [inline src/appendix/git.md]: @Charlotteis thanks again.

File: src/appendix/git.md (hunk 5 above)
Comment (dvberkel) [inline src/appendix/git.md:33]: @Charlotteis I am using the [reference style of linking](https://daringfireball.net/projects/markdown/syntax#link) here, and the shorthand variant on top of that.

The full form would be `[contributing][contributing]`. When the reference is the same as what is displayed the reference can be omitted.

You still need to point to an actual resource. I have collected several reference at the end of the document. For `contributing` I added

``` text
[contributing]: https://github.com/intermezzOS/book
```

File: src/appendix/git.md (hunk 6 above)
Comment (dvberkel) [inline src/appendix/git.md]: @Charlotteis this is a very good point. Would you suggest to alter the flow of the text or add an "Help I already added a file that I want to ignore"-section?

File: src/appendix/git.md (hunk 5 above)
Comment (varjmes) [inline src/appendix/git.md:33]: > You still need to point to an actual resource. I have collected several reference at the end of the document. For contributing I added

I get it, thanks :)

File: src/appendix/git.md (hunk 6 above)
Comment (varjmes) [inline src/appendix/git.md]: I'd alter the flow, placing 'Ignoring' before 'Adding'

Comment (dvberkel) [inline src/appendix/git.md]: Ok, I will work on that. Thanks for the feedback

**Comment by steveklabnik** [comment]
Thanks a ton for this! I'm sorry it's taken me a while to review, I haven't had time. I'm hoping to early next week.

**Comment by dvberkel** [comment]
@steveklabnik don't worry. I fully understand that time is very precious. And that you need to divide it between a lot of things.

As per the feedback of @Charlotteis I will change the order of the sections _adding_ and _ignoring_. I will probably finish that tomorrow.

File: src/appendix/git.md (hunk 6 above)
Comment (dvberkel) [inline src/appendix/git.md]: @Charlotteis I changed the order of the sections 'Ignoring' and 'Adding'

**Comment by steveklabnik** [comment]
Oh my! After far too long, let's get this merged. So sorry about that :(

**Comment by dvberkel** [comment]
@steveklabnik It is OK, no hard feelings. I am fully aware that everybody has a complex schedule and that you can spend your time only once. Just make sure to stay happy and healthy

**Pull request merged**